src/libinjector/linux/linux_utils.c

```diff
@@ -0,0 +1,92 @@
+#include "linux_utils.h"
+#include "linux_debug.h"
+#include <sys/mman.h>
+#include <fcntl.h>
+
+bool setup_exit_syscall(injector_t injector, x86_registers_t* regs, uint32_t no) {
+    // exit (exit_no)
+    struct argument args[10] = { {0} };
+    init_int_argument(&args[0], no);
+
+    bool success = setup_stack(injector->drakvuf, regs, args, ARRAY_SIZE(args));
+    if(success)
+        injector->syscall = sys_exit;
+
+    return success;
+}
+
+bool check_userspace_int3_trap(injector_t injector, drakvuf_trap_info_t* info) {
+
+    // check CPL
+    unsigned long int CPL = (info->regs->cs_sel & 3);
+    PRINT_DEBUG("CPL 0x%lx\n", CPL);
+
+    if ( CPL != 0)
+    {
+        PRINT_DEBUG("Reached userspace, yayy!\n");
+    }
+    else
+    {
+        PRINT_DEBUG("INT3 received but CPL is not 0x3\n");
+        print_stack(injector->drakvuf, info);
+        print_registers(info);
+        return false;
+    }
+
+    if ( info->proc_data.pid != injector->target_pid )
+    {
+        PRINT_DEBUG("INT3 received but '%s' PID (%u) doesn't match target process (%u)\n",
+                    info->proc_data.name, info->proc_data.pid, injector->target_pid);
+        return false;
+    }
+
+    if (info->regs->rip != info->trap->breakpoint.addr) {
```

Comment:
is it possible?

Reply:
found the check in `win_injector` so not sure

Reply:
My guess would be that it shouldn't happen. Still I think assert would fit better if we are "almost" sure.